### ISSUE TITLE
barh edgecolor and hatch are not applied to all bars

### ISSUE THREAD

**Issue by rpshaw** [matplotlib/matplotlib#9496]
### Bug report

**Bug summary**

When I specify `edgecolor` and `hatch` within `pyplot.barh` those settings are only applied to the first bar in the plot.

**Code for reproduction**

```
import matplotlib.pyplot as plt

values = [13, 2, 1, 1]
y_pos = [0, 1, 2, 3]
fig = plt.figure(figsize=(6.0, 3.3))
ax = fig.add_subplot(111)
ax.barh(y_pos,
        values,
        align="center",
        facecolor="white",
        edgecolor="blue",
        hatch="/",
        )
ax.set_yticks(y_pos)
ax.set_yticklabels([])
plt.show()

```

**Actual outcome**

![test_barh](https://user-images.githubusercontent.com/11755899/31841465-cd9a2e42-b5a6-11e7-8d20-7fbc77e1a669.png)

**Expected outcome**

All bars to show the blue `edgecolor` and `hatch`.

This worked in Matplotlib 2.0.1.

**Matplotlib version**

  * Operating system: Red Hat Enterprise Linux Workstation release 6.8 (Santiago)
  * Matplotlib version: 2.1.0
  * Matplotlib backend: 'Qt5Agg'
  * Python version: 3.6.2
  * Jupyter version (if applicable): N/A
  * Other libraries: 

I installed this using the main channel of miniconda.

**Comment by afvincent** [comment]
Thank you @rpshaw for reporting this issue. It seems to have been fixed on the master branch by #9353. Here is what I get when executing your example with a freshly fetched master:
![mwe_9496](https://user-images.githubusercontent.com/17270724/31841838-20a1caf2-b5a0-11e7-900d-eacb8f4207d7.png).

Closing, but feel free to reopen if you think that I am missing something :).

**Comment by rpshaw** [comment]
Thanks for letting me know!

**Comment by QuLogic** [comment]
This is a duplicate of #9351 and was fixed by #9353 so should be in 2.1.1 when it's released.

**Comment by QuLogic** [comment]
Oops, sorry @afvincent; didn't notice you mentioned that already.
